File: Angular_Frontend/src/app/models/Port.ts
/// <summary>
/// Represents the configuration details of a serial port.
/// </summary>
export class Port
{
    PortName:string="COM6"
    BaudRate:number=9600
    DataBits:number=8
    Handshake:string="None"
    StopBits:string="One"
    Parity:string="None"

    /// <summary>
    /// Constructor to initialize the serial port configuration.
    /// </summary>
    /// <param name="ptname">Name of the port</param>
    /// <param name="bdrate">Baud rate</param>
    /// <param name="dtbits">Data bits</param>
    /// <param name="hshake">Handshake setting</param>
    /// <param name="stbits">Stop bits</param>
    /// <param name="par">Parity setting</param>
    constructor(ptname:string,bdrate:number,dtbits:number,hshake:string,stbits:string,par:string)
    {
        this.PortName=ptname
        this.BaudRate=bdrate
        this.DataBits=dtbits
        this.Handshake=hshake
        this.StopBits=stbits
        this.Parity=par
    }
}

File: DotNet_Backend/DotNet_Backend/TestWebApi/Models/Port.cs
using System.IO.Ports;
using System.Text.Json.Serialization;

namespace TestWebApi.Models
{
	/// <summary>
	/// This class is used to transfer the connection paremeters for Serial Port connection 
	/// from the frontend to backend.
	/// </summary>
	public class Port
	{
		[JsonPropertyName("PortName")]
		public string PortName { get; set; }
		public int BaudRate { get; set; }
		public int DataBits { get; set; }
		public string Handshake { get; set; }
		public string StopBits { get; set; }
		public string Parity { get; set; }
	}
}
